### DISCUSSION TITLE
端口打开错误

### DISCUSSION THREAD

**Discussion by enzo1985** [rerender2021/language-shadow#29] [General]
你好，我使用是绑定端口有问题
我用netstat -ano | findstr 8100  查看并没有被占用，防火墙也是关闭的
![1681545983488](https://user-images.githubusercontent.com/4523988/232199193-3903cf7a-826d-4cac-9ea7-d0326432e0e4.png)

**Comment by rerender2021** [comment]
- 在启动的时候会弹个窗口问要不要允许使用xxx网络之类的（public/private），这个当时是点的允许吗？
- 另外我查了一下：
![image](https://user-images.githubusercontent.com/87927336/232260630-0a0e3fd0-1064-472d-8a37-0bd1c5fc5cc7.png)
![image](https://user-images.githubusercontent.com/87927336/232260637-5bf8548b-f338-4c35-b233-13545ef7da2b.png)

所以需要额外检查一下：
- 有没有多开
- 有没有杀毒软件
- 重启一下
- 实在不行之后支持自定义端口以避免占用问题：https://github.com/rerender2021/language-shadow/issues/26

**Reply by enzo1985** [reply]
感觉还是像程序的问题，是下载zip文件直接运行exe，不用装其他依赖吧
我重启电脑后，立马运行debug的程序，发现还是无法绑定socket
没有开杀毒软件和防火墙，netstat 也并没有发现端口被占用，很奇怪

**Reply by rerender2021** [reply]
“不用装其他依赖”：是的，不用。

注意到报错中中文描述是乱码，可以这样尝试一下吗？把系统语言切换为英文，然后再运行一下，看一下乱码部分对应的报错内容是什么，有没有更多的线索。

**Reply by rerender2021** [reply]
还有其它可能原因：端口被用作了系统默认端口
- win10下无进程占用端口但是又提示端口占用(实际没有)：https://blog.csdn.net/kkun/article/details/109691064
- 详细查看方式：https://zhuanlan.zhihu.com/p/511925609

**Comment by enzo1985** [comment]
我还是等你的下一个可以配置端口的版本吧

rerender2021 ***@***.***> 于2023年4月17日周一 22:49写道：

> 还有其它可能原因：
>
>    - https://blog.csdn.net/kkun/article/details/109691064
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rerender2021/language-shadow/discussions/29#discussioncomment-5638188>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCQPVCME5GFNWGHLQYYUC3XBVKBDANCNFSM6AAAAAAW7IHKEY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Best Regards
enzofang

**Reply by rerender2021** [reply]
好的

**Reply by rerender2021** [reply]
> 我还是等你的下一个可以配置端口的版本吧 rerender2021 ***@***.***> 于2023年4月17日周一 22:49写道：
> […](#)
> 还有其它可能原因： - https://blog.csdn.net/kkun/article/details/109691064 — Reply to this email directly, view it on GitHub <[#29 (reply in thread)](https://github.com/rerender2021/language-shadow/discussions/29#discussioncomment-5638188)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABCQPVCME5GFNWGHLQYYUC3XBVKBDANCNFSM6AAAAAAW7IHKEY> . You are receiving this because you authored the thread.Message ID: ***@***.*** com>
> -- Best Regards enzofang

实现好了，可以试试：https://github.com/rerender2021/language-shadow/releases/tag/preview%2Fcustom-port

**Comment by enzo1985** [comment]
试了下，可以用了

rerender2021 ***@***.***> 于2023年4月18日周二 22:06写道：

> 我还是等你的下一个可以配置端口的版本吧 rerender2021 *@*.
>
> *> 于2023年4月17日周一 22:49写道： … <#m_-5795564048352780067_> 还有其它可能原因： -
> https://blog.csdn.net/kkun/article/details/109691064
> <https://blog.csdn.net/kkun/article/details/109691064> — Reply to this
> email directly, view it on GitHub <#29 (reply in thread)
> <https://github.com/rerender2021/language-shadow/discussions/29#discussioncomment-5638188>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ABCQPVCME5GFNWGHLQYYUC3XBVKBDANCNFSM6AAAAAAW7IHKEY
> <https://github.com/notifications/unsubscribe-auth/ABCQPVCME5GFNWGHLQYYUC3XBVKBDANCNFSM6AAAAAAW7IHKEY>
> . You are receiving this because you authored the thread.Message ID: @.*
> com>
> -- Best Regards enzofang
>
> 实现好了，可以试试：
> https://github.com/rerender2021/language-shadow/releases/tag/preview%2Fcustom-port
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rerender2021/language-shadow/discussions/29#discussioncomment-5649584>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCQPVBWMWCETKOTHJBUZLLXB2NV5ANCNFSM6AAAAAAW7IHKEY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Best Regards
enzofang

**Reply by rerender2021** [reply]
好的
